pkg/storage/gcp/bucket_cloud.go

```diff
@@ -56,14 +56,12 @@ func (b *gcpBucket) List(ctx context.Context, prefix string) ([]string, error) {
 	return res, nil
 }
 
-func (b *gcpBucket) Exists(ctx context.Context, path string) bool {
+func (b *gcpBucket) Exists(ctx context.Context, path string) (bool, error) {
+	const op errors.Op = "gcpBucket.Exists"
 	_, err := b.Object(path).Attrs(ctx)
-	if err == nil {
-		return true
-	} else if err == storage.ErrObjectNotExist {
```

Comment:
We should probably still differentiate between the object not existing and another error trying to check the object attributes. I guess the caller of `Exists` can be doing that however so ignore me here.

Reply:
Actually I totes agree with @robjloranger, Exists should return false with *no error* if err == storage.ErrObjectNotExist. Because we got an honest answer, and not an error. It was just communicated through an error variable.